### PR TITLE
Updated RS Hurl Script to Replace MSH Header When Not Toggled

### PR DESCRIPTION
**Pull request by pluckyswan**
# Replace MSH Header When Not Toggled

Added so by default when hurl script for ReportStream is ran, MSH header is replaced to prevent out bound to a real receiver. Added a flag to allow out bound in command line. 

## Issue

https://github.com/CDCgov/trusted-intermediary/issues/1322

## Checklist

- [x] I have added logging where useful (with appropriate log level)
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: scripts/hurl/rs/hrl [hunk 1]
```diff
@@ -29,6 +30,7 @@ show_help() {
     echo "    -i <SUBMISSION_ID>    The submissionId to call the history API with (Required for history API)"
     echo "    -v                    Verbose mode"
     echo "    -h                    Display this help and exit"
+    echo "    -p                    Produce a real message, receiver will not be Flexion specific"
```
Comment (basiliskus) [inline scripts/hurl/rs/hrl:33]: It's not clear to me what this flag is doing given the help message. Is the flag just scrambling the receiver facility details?

File: scripts/hurl/rs/hrl [hunk 2]
```diff
@@ -114,6 +119,21 @@ if [ -z "$secret" ]; then
...
 fi
 
+
+if [ "$allow_outbound" = false ]; then
+  file_path=$(dirname "$fpath")
+  substitutedFilePathPrefix="$CDCTI_HOME/substituted_receivers"
+#  Make temporary directory to store scrambled file
+  mkdir -p "$substitutedFilePathPrefix/$file_path" && touch -f "$substitutedFilePathPrefix/$fpath"
+#  Grab MSH Header receiver fields
+  receiverFieldsToSubstitute=$(head -n1 "$CDCTI_HOME/$fpath" | cut -d "|" -f 5,6)
```
Comment (basiliskus) [inline scripts/hurl/rs/hrl:129]: Instead of replacing MSH-5/6, we can replace the whole MSH segment. That will also handle some other edge cases. For orders, we can replace it with: `MSH|^~\&|Sender Application^sender.test.com^DNS|Sender Facility^0.0.0.0.0.0.0.0^ISO|Receiver Application^0.0.0.0.0.0.0.0^ISO|Receiver Facility^simulated-lab-id^DNS|20230101010000-0000||ORM^O01^ORM_O01|111111|T|2.5.1||||||||||` and for results:  `MSH|^~\&|Sender Application^sender.test.com^DNS|Sender Facility^0.0.0.0.0.0.0.0^ISO|Receiver Application^0.0.0.0.0.0.0.0^ISO|Receiver Facility^simulated-hospital-id^DNS|20230101010000-0000||ORU^R01^ORU_R01|111111|T|2.5.1||||||||||`

File: scripts/hurl/rs/hrl (hunk 1 above)
Comment (pluckyswan) [reply scripts/hurl/rs/hrl:33]: Description updated to be more concise. Thoughts?

Comment (basiliskus) [reply scripts/hurl/rs/hrl:33]: The line above also needs updating. I think the update is good, but I'd remove the `Produce a real message` part which can create confusion. I'd also add something explaining the default behavior. It could be something like: `By default, the MSH segment is replaced to contain a Flexion receiver in order to avoid sending messages to partners inadvertently. When using this flag, the MSH segment will not be replaced`

File: scripts/hurl/rs/hrl (hunk 2 above)
Comment (pluckyswan) [reply scripts/hurl/rs/hrl]: Implemented suggestions as discussed!

File: scripts/hurl/rs/hrl (hunk 1 above)
Comment (pluckyswan) [reply scripts/hurl/rs/hrl]: I added this to the readme and option description.

File: .gitignore [hunk 3]
```diff
@@ -34,3 +34,6 @@ __pycache__
 localfileorder.json
 
 logback.log
+
+# Ignore folder created when substituting receiver fields in MSH Header
+substituted_receivers
```
Comment (basiliskus) [inline .gitignore:39]: Looks like this folder was renamed to `substituted_header`?

Comment (basiliskus) [reply .gitignore:39]: Can the temporary file be deleted after calling the hurl command instead?

Comment (pluckyswan) [reply .gitignore]: I implemented your suggestion

File: scripts/hurl/rs/hrl [hunk 4]
```diff
@@ -114,6 +120,30 @@ if [ -z "$secret" ]; then
...
+  msh_header=$(head -n1 "$CDCTI_HOME/$fpath")
+# Check message type
+  if [[ "$msh_header" = *"^R01"* ]]; then
+    msh_header_replacement="MSH|^~\&|Sender Application^sender.test.com^DNS|Sender Facility^0.0.0.0.0.0.0.0^ISO|Receiver Application^0.0.0.0.0.0.0.0^ISO|Receiver Facility^simulated-hospital-id^DNS|20230101010000-0000||ORU^R01^ORU_R01|111111|T|2.5.1||||||||||"
+  elif [[ "$msh_header" = *"^O01"* ]]; then
+    msh_header_replacement="MSH|^~\&|Sender Application^sender.test.com^DNS|Sender Facility^0.0.0.0.0.0.0.0^ISO|Receiver Application^0.0.0.0.0.0.0.0^ISO|Receiver Facility^simulated-lab-id^DNS|20230101010000-0000||ORM^O01^ORM_O01|111111|T|2.5.1||||||||||"
+  elif [[ "$msh_header" = *"^O21"* ]]; then
+      msh_header_replacement="MSH|^~\&|Sender Application^sender.test.com^DNS|Sender Facility^0.0.0.0.0.0.0.0^ISO|Receiver Application^0.0.0.0.0.0.0.0^ISO|Receiver Facility^simulated-lab-id^DNS|20230101010000-0000||OML^O21^OML_O21|111111|T|2.5.1||||||||||"
+  else
+      echo "File does not contain any valid subfields."
```
[the remark this answers is not in the document]
Comment (pluckyswan) [reply scripts/hurl/rs/hrl]: Updated it
